modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/odata/ExpressionVisitorImpl.java

```diff
@@ -338,6 +350,58 @@ private String _getPropertyName(Object object) {
 		return String.valueOf(object);
 	}
 
+	private Object _getSub(Object left, Object right) {
+		Duration duration = Duration.parse(right.toString());
```

Comment:
Maybe the tight is not a duration, you need to check first that the values are correct and supported, and them convert them.